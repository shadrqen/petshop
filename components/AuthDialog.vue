<template>
  <base-dialog
    :dialog="dialog"
    :close-dialog="closeDialog"
    :max-width="420"
  >
    <template #title>
      <v-avatar>
        <img
          :src="require('@/assets/logo.PNG')"
          alt="Logo"
        >
      </v-avatar>
    </template>
    <template #content>
      <v-form
        :ref="formRef"
        @submit.prevent="submitContent"
      >
        <slot name="text-fields" />
        <v-checkbox
          :data-test-id="checkboxDataTestId"
          color="#50C494FF"
          :label="checkboxLabel"
          class="mt-n2 mb-2"
          required
        />
        <base-button
          width="100%"
          color="white"
          class="white--text"
          :background-color="btnBackgroundColor"
          :data-test-id="dataTestId"
          :btn-id="actionBtnId"
          :btn-class="actionBtnClass"
          :action-ongoing="submissionOngoing"
          :submit-content="submitContent"
        >
          <template #button-body>
            <div
              v-if="submissionOngoing"
              class="lds-ellipsis"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
            <span
              v-else
              class="text-button"
              v-text="actionName"
            />
          </template>
        </base-button>
      </v-form>
    </template>
    <template #action-buttons>
      <slot name="action-buttons" />
    </template>
  </base-dialog>
</template>

<script>
import BaseDialog from '@/components/Base/BaseDialog'
import BaseButton from '@/components/Base/BaseButton'
export default {
  name: 'AuthDialog',
  components: {
    BaseDialog,
    BaseButton
  },
  props: {
    closeDialog: {
      type: Function,
      required: true
    },
    dialog: {
      type: Boolean,
      required: true
    },
    submitContent: {
      type: Function,
      required: true
    },
    dataTestId: {
      type: String,
      required: true
    },
    actionBtnId: {
      type: String,
      required: false,
      default: ''
    },
    actionBtnClass: {
      type: String,
      required: false,
      default: ''
    },
    formRef: {
      type: String,
      required: true
    },
    actionName: {
      type: String,
      required: true
    },
    submissionOngoing: {
      type: Boolean,
      required: true,
      default: false
    },
    checkboxLabel: {
      type: String,
      required: true
    },
    checkboxDataTestId: {
      type: String,
      required: true
    },
    error: {
      type: Object,
      required: true
    }
  },
  data () {
    return {
      btnBackgroundColor: '#50c494'
    }
  }
}
</script>

<style lang="scss" scoped>
</style>
