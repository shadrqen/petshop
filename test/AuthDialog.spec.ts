import { createLocalVue, shallowMount } from '@vue/test-utils'
import AuthDialog from '~/components/Auth/AuthDialog.vue'

describe('~/components/Auth/AuthDialog.vue', () => {
  const localVue = createLocalVue()

  it('should render component and match snapshot', () => {
    const { element } = shallowMount(AuthDialog, {
      localVue,
      propsData: {
        dialog: true,
        dataTestId: 'someId',
        formRef: 'someRef',
        actionName: 'someAction',
        checkboxLabel: 'someLabel',
        checkboxDataTestId: 'someDataId',
        submissionOngoing: false,
        error: {
          status: false
        }
      }
    })

    expect(element.innerHTML).toMatchSnapshot()
  })
})
