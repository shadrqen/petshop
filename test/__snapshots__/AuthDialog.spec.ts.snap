// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`~/components/Auth/AuthDialog.vue should render component and match snapshot 1`] = `
"<v-row-stub tag=\\"div\\" class=\\"mt-n6\\"><v-col-stub tag=\\"div\\"><v-row-stub tag=\\"div\\" class=\\"justify-center\\"><base-avatar-stub avatarsize=\\"50\\" avatarclass=\\"mx-2\\" imagesrc=\\"logo.PNG\\" imagealt=\\"Logo\\"></base-avatar-stub></v-row-stub> <v-row-stub tag=\\"div\\" class=\\"justify-center mt-6 mb-3\\">
          someAction
        </v-row-stub></v-col-stub></v-row-stub><v-form-stub> <v-checkbox-stub color=\\"#50C494FF\\" errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" label=\\"someLabel\\" ripple=\\"true\\" valuecomparator=\\"[Function]\\" indeterminateicon=\\"$checkboxIndeterminate\\" officon=\\"$checkboxOff\\" onicon=\\"$checkboxOn\\" data-test-id=\\"someDataId\\" required=\\"\\" class=\\"mt-n2 mb-2\\"></v-checkbox-stub> <!----> <base-button-stub btnid=\\"\\" btnclass=\\"\\" datatestid=\\"someId\\" width=\\"100%\\" color=\\"white\\" backgroundcolor=\\"#50c494\\" texttransform=\\"\\" class=\\"white--text\\"></base-button-stub></v-form-stub>"
`;
